### ISSUE TITLE
selectionRequired

### ISSUE THREAD

**Issue by ezdavis** [sergiodlopes/jquery-flexdatalist#96]
I'm having an issue with selectionRequired not doing anything. I'm getting my data from a json file and everything is working fine except if I leave the field blank when clicking my submit it doesn't stop it from submitting. I would like for it to prompt for input if there isn't a value or if there is a value but they haven't selected one of the options from the dropdown. I'm using "valueProperty='*'" so I need them to select from the options in order to attach all my data points.

**Comment by ezdavis** [comment]
Here is my input tag
`<input type="text" id="PlayerName" placeholder="Player Name" class="form-control flexdatalist" data-data='json/player_results.json' 
										data-search-in='name' data-visible-properties='["name", "pos", "team", "bye"]' data-value-property='*' 
										data-selection-required='true' data-text-property='{name}, {pos}, {team}' data-min-length='0' name="bid[0][playerInfo]:object" >`

**Comment by sergiodlopes** [comment]
Hi,

Maybe I misunderstood your problem, but, what you're asking is to flexdatalist do the input validation (prevent form submittion if empty)?

Sérgio

**Comment by ezdavis** [comment]
Yes but not just prevent submission if empty. I also want to prevent submission if they haven't selected one of the items from the dropdown. For instance if the I have "John Doe" in the dropdown I don't want them to be able to type "Joh" and submit. I thought this is what the data-selection-required tag would do.

**Comment by ezdavis** [comment]
I'm expecting the result from the example for "country_name_required_selection" where you can begin typing and once you click off or tab it clears the field back out if they didn't select one of the items from the dropdown. This isn't happening in my case though.

**Comment by ezdavis** [comment]
So i figured out the problem. I can get it to work if I remove "data-value-property='*'".  If I remove this I can type in the field and when I click off of it without selecting an item from the dropdown it goes away. Could you look into this?

**Comment by ezdavis** [comment]
Any idea?

**Comment by ezdavis** [comment]
Still looking for a solution to this issue. Any chance you could look into it?

**Comment by ezdavis** [comment]
I was able to find the issue and fix it. When using "data-value-property='*'" it brings back an object. Because of this I needed to check the size of the object. Below is what I changed.
Original:
'/**
         * Clear text/alias input when criteria is met.
         */
            clearText: function (event) {
                var val = _this.fvalue.get(),
                    options = _this.options.get();

                if (!options.multiple && options.selectionRequired && val.length === 0) {
                    $alias[0].value = '';
                }
            },'
Modified:
'/**
         * Clear text/alias input when criteria is met.
         */
            clearText: function (event) {
                var val = _this.fvalue.get(),
                    options = _this.options.get();

                if (!options.multiple && options.selectionRequired && (val.length === 0 || Object.keys(val).length === 0)) {
                    $alias[0].value = '';
                }
            },'

**Comment by fabsahe** [comment]
Hi, I have the same problem, is this the solution?
